File: wk3/cpp/tripletexting.cpp
// problem link: https://open.kattis.com/problems/tripletexting

#include <bits/stdc++.h>
using namespace std;

int main()
{
    ios::sync_with_stdio(false); cin.tie(0); cout.tie(0);
    
    string t;
    cin >> t;
    string a = t.substr(0, t.size() / 3), b = t.substr(t.size() / 3, t.size() / 3), c = t.substr(2 * (t.size() / 3), 2 * (t.size() / 3));
    if(a == b || a == c)
        cout << a;
    else
        cout << b;
    
    return 0;
}

File: wk3/java/tripletexting.java
// problem link: https://open.kattis.com/problems/tripletexting

import java.util.*;
import java.io.*;
public class tripletexting {
    public static void main(String[] args) throws IOException {
        BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
        String t = r.readLine(), a = t.substring(0, t.length() / 3), b = t.substring(t.length() / 3, 2 * (t.length() / 3)), c = t.substring(2 * (t.length() / 3), t.length());
        if(a.equals(b) || a.equals(c))
            System.out.println(a);
        else
            System.out.println(b);
    }
}
